data-entry/dataEntry.app.js

```diff
@@ -78,12 +78,62 @@
                         if (key.simple) {
                             (function(key) {
                                 key.getDomainValues().then(function success(values) {
-                                    dataEntryModel.domainValues[key.colset.columns[0].name] = [];
-                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name];
-
-                                    angular.forEach(values.data, function(value) {
-                                        var field = Object.keys(value)[0];
-                                        domainValues.push(value[field]);
+                                    var domainValues = dataEntryModel.domainValues[key.colset.columns[0].name] = [];
+                                    var table = key.key._table;
+                                    var annotations = table.annotations.all();
+                                    var termColumnName = null;
+
+
+                                    // TODO need to figure out how to handle if mutiple annotations are present
+                                    angular.forEach(annotations, function(annotation) {
+                                        // there should only be one term annotation
+                                        if (annotation.content.term) {
```

Comment:
You cannot check on the payload of an annotation like this. There is no uniqueness guarantee for `term`. The test must be on the unique annotation key, like `tag...`.